### PR TITLE
Chanakya/redeem dust value

### PR DESCRIPTION
**Pull request by Chanakya888**
I let the `onSubmit` check intact, just in case any user inputs the value from console

### CODE REVIEW

File: src/pages/Application/redeem/enter-amount-and-address.tsx [hunk 1]
```diff
@@ -211,6 +211,11 @@ function EnterAmountAndAddress(): ReactElement {
     setUsdAmount(getUsdAmount(amount, usdPrice));
     const fee = await window.polkaBTC.redeem.getFeesToPay(amount);
     setRedeemFee(fee);
+
+    if (Number(amount) <= Number(dustValue)) {
+      setAmountPolkaBTC('0.00');
+      toast.warning(t('redeem_page.amount_greater') + dustValue + 'BTC)');
```
Comment (nud3l) [inline src/pages/Application/redeem/enter-amount-and-address.tsx:217]: This error message should be displayed below the input form just like on redeem. Not a toast.

Also, the amount should not be set if it fails the validation, e.g. the `setAmmountPolkaBTC` should not be called.

**Review by nud3l — CHANGES_REQUESTED** [review]

File: src/pages/Application/redeem/enter-amount-and-address.tsx (hunk 1 above)
Comment (nud3l) [reply src/pages/Application/redeem/enter-amount-and-address.tsx:217]: @anton-karlovskiy I think we could also have a general "input validation" function that is applicable for both issue and redeem. We have several checks in common like:

Input amount
- dust value check

State of the app
- parachainStatus is Running
- address is set
- bitcoin height (e.g. https://github.com/interlay/polkabtc-ui/blob/dev/src/pages/Application/issue/enter-btc-amount.tsx#L221 and https://github.com/interlay/polkabtc-ui/blob/dev/src/pages/Application/redeem/enter-amount-and-address.tsx#L138)

Comment (Chanakya888) [reply src/pages/Application/redeem/enter-amount-and-address.tsx:217]: 1. Okay, I'll add the error message below the field.
2.  for clarification, my assumption of `disallow` in the ticket meant the user shouldn't be allowed to enter an amount that is less than the `dust value` which is what I tried to do here, whenever a user enters a value that is less than dust value, the value itself changes to 0 and throws in error, am I in the right direction here?

Comment (nud3l) [reply src/pages/Application/redeem/enter-amount-and-address.tsx:217]: Yep, the code to display error messages below the input field is already there so should hopefully be easy to add.

If the user enters an amount below the dust value, the error field below the input field is displayed and the value changes to `undefined`

Comment (Chanakya888) [reply src/pages/Application/redeem/enter-amount-and-address.tsx]: wait a sec, sorry (I'm a bit confused as I am not able to test it). You'd want the input field value to change to `0.00` (not `undefined`) if value < dustValue

Comment (nud3l) [reply src/pages/Application/redeem/enter-amount-and-address.tsx]: The input field itself should stay at whatever was just entered by the user. The amount itself should be undefined.

Please have a look how this is done in issue. We don't need to introduce any booleans component level variables for this: https://github.com/interlay/polkabtc-ui/blob/dev/src/pages/Application/issue/enter-btc-amount.tsx#L279

**Comment by Chanakya888** [comment]
@nud3l Closing this PR confirmed with Dan that he wanted a check onSubmit, @anton-karlovskiy says he will give real time validation a try during his refactoring.

**Pull request closed without merging**